src/v/kafka/server/replicated_partition.h

```diff
@@ -44,6 +44,10 @@ class replicated_partition final : public kafka::partition_proxy::impl {
 
     bool is_leader() const final { return _partition->is_leader(); }
 
+    ss::future<result<model::offset>> linearizable_barrier() {
+        return _partition->linearizable_barrier();
```

Comment:
This needs translation to kafka offset with `_translator->from_log_offset`. All offsets exposed by `replicated_partition` are in Kafka offset domain.

Reply:
great catch!